core/domain/acl_decorators.py

```diff
@@ -126,3 +167,320 @@ def test_can_view_stats(self, exploration_id, **kwargs):
+            return
+        exploration_rights = rights_manager.get_exploration_rights(
+            exploration_id)
+
+        if exploration_rights is None:
+            raise self.PageNotFoundException
+
+        if check_exploration_editable(
+                self.user_id, self.actions, exploration_rights):
+            return handler(self, exploration_id, **kwargs)
+        else:
+            raise self.UnauthorizedUserException(
+                'You do not have the credentials to manage suggestions for ' +
+                'this exploration.')
+
+    return test_can_manage
+
+
+def can_publish_exploration(handler):
+    """Decorator to check whether user can publish exploration."""
+
+    def test_can_publish(self, exploration_id, **kwargs):
+        exploration_rights = rights_manager.get_exploration_rights(
+            exploration_id)
+
+        if exploration_rights is None:
+            raise self.PageNotFoundException
+
+        if exploration_rights.cloned_from:
+            raise self.UnauthorizedUserException(
+                'You do not have credentials to publish this exploration.')
+
+        if role_services.ACTION_PUBLISH_ANY_EXPLORATION in self.actions:
+            return handler(self, exploration_id, **kwargs)
+
+        if exploration_rights.status == rights_manager.ACTIVITY_STATUS_PRIVATE:
+            if role_services.ACTION_PUBLISH_OWNED_EXPLORATION in self.actions:
+                if self.user_id in exploration_rights.owner_ids:
+                    return handler(self, exploration_id, **kwargs)
+
+        if exploration_rights.status == rights_manager.ACTIVITY_STATUS_PUBLIC:
+            if role_services.ACTION_PUBLISH_PUBLIC_EXPLORATION in self.actions:
+                return handler(self, exploration_id, **kwargs)
+
+        raise self.UnauthorizedUserException(
+            'You do not have credentials to publish this exploration.')
+
+    return test_can_publish
+
+
+def can_modify_exploration_roles(handler):
+    """Decorators to check whether user can manage rights related to an
+    exploration.
+    """
+
+    def test_can_modify(self, exploration_id, **kwargs):
+
+        exploration_rights = rights_manager.get_exploration_rights(
+            exploration_id)
+
+        if exploration_rights is None:
+            raise self.PageNotFoundException
+
+        if (
+                exploration_rights.community_owned or
+                exploration_rights.cloned_from):
+            raise self.UnauthorizedUserException(
+                'You do not have credentials to change rights for this' +
+                'exploration.')
+
+        if (
+                role_services.ACTION_MODIFY_ROLES_FOR_ANY_EXPLORATION in
+                self.actions):
+            return handler(self, exploration_id, **kwargs)
+        if (
+                role_services.ACTION_MODIFY_ROLES_FOR_OWNED_EXPLORATION in
+                self.actions):
+            if self.user_id in exploration_rights.owner_ids:
```

Comment:
oh sorry. I misinterpreted the comment.

Yes, sounds a good idea. Added is_owner, is_editor and is_viewer